var/spack/repos/builtin/packages/lcio/package.py

```diff
@@ -72,3 +72,8 @@ def url_for_version(self, version):
             print('Error - Wrong version format provided')
             return
         return url
+        
+    def setup_dependent_run_environment(self, env, dependent_spec):
```

Comment:
This adds these env vars to the module files of any package that depends on `lcio`, are you sure you don't want `setup_run_environment` instead?

Reply:
@adamjstewart Thanks for the comments! Yes, I think you are right, but I'm not 100% sure I understand the difference. does it mean that if I do `spack load PackageDependingOnLCIO`, `$LCIO` will be set, also if I use setup_run_environment?

Reply:
This relates to the discussion in #8639. Basically, you can modify your `modules.yaml` to autoload run dependencies (I _think_ this also applies to `spack load` which no longer uses modules), but by default it will not be set.